File: frontend-vue/src/types/shipment.ts
export interface Route {
  id: number
  source: string
  destination: string
  averageDuration: number
}

export interface Customer {
  id: number
  name: string
  email: string
  phone: string
}

export interface Shipment {
  id: number
  trackingNo: string
  sender: string
  receiver: string
  routeId: number
  route?: Route
  customerId?: number
  customer?: Customer
  weight: number
  status: string
  createdAt: string
  riskScore?: number
  riskLevel?: string
}

export interface CreateShipmentInput {
  trackingNo: string
  sender: string
  receiver: string
  routeId: number
  customerId: number
  weight: number
  status: string
}

File: backend-net/SmartLogix.WebApi/Models/Shipment.cs
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace SmartLogix.WebApi.Models
{
    public class Shipment
    {
        public int Id { get; set; }
        public string TrackingNo { get; set; } = string.Empty;
        public string Sender { get; set; } = string.Empty;
        public string Receiver { get; set; } = string.Empty;
        
        public int CustomerId { get; set; }
        public Customer? Customer { get; set; }

        public int RouteId { get; set; }
        public Route? Route { get; set; }

        public decimal Weight { get; set; }
        public string Status { get; set; } = "Pending"; // Pending, InTransit, Delivered, Delayed
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        [JsonIgnore]
        public ICollection<RiskScore> RiskScores { get; set; } = new List<RiskScore>();
    }
}
